html/user/account_ownership.php

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env php
+<?php
+// This file is part of BOINC.
+// http://boinc.berkeley.edu
+// Copyright (C) 2008 University of California
+//
+// BOINC is free software; you can redistribute it and/or modify it
+// under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation,
+// either version 3 of the License, or (at your option) any later version.
+//
+// BOINC is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+// See the GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with BOINC.  If not, see <http://www.gnu.org/licenses/>.
+
+require_once("../inc/boinc_db.inc");
+require_once("../inc/user.inc");
+require_once("../inc/util.inc");
+require_once("../inc/countries.inc");
+require_once('../inc/recaptchalib.php');
+
+check_get_args(array("tnow", "ttok"));
+
+$user = get_logged_in_user();
+check_tokens($user->authenticator);
+
+function account_ownership_action($user) {
+  // POST request - the user has submitted the form.
+  page_head(tra("Proof of account ownership results"), null, null, null, boinc_recaptcha_get_head_extra());
+
+  global $recaptcha_private_key;
+  if ($recaptcha_private_key) {
+      // Recaptcha is enabled on the BOINC server
+      if (!boinc_recaptcha_isValidated($recaptcha_private_key)) {
+          // The user failed to solve the recaptcha prompt - redirect them to an error message!
+          error_page(
+              tra("Your reCAPTCHA response was not correct. Please try again.")
+          );
+      }
+  }
+
+  // Input is passed in from the openssl_sign_form
+  $user_data = htmlentities(post_str("user_data", true), ENT_QUOTES, "UTF-8"); // Convert special characters to html equivelant
+
+  if ((strlen($user_data) > 0) && (strlen($user_data) <= 4096)) {
+      // The user data input is valid
+      $config = get_config();
+      $keydir = parse_config($config, "<key_dir>");
+
+      // If the following keys do not exist, then the users will be shown an error message.
+      $private_key_path = "file://$keydir/account_ownership_private.pem";
+      $public_key_path = "file://$keydir/account_ownership_public.pem";
```

Comment:
These four lines are repeated .  As a result, this logic should be moved into an include file (maybe create inc/account_ownership.inc?)